internal/jennies/terraform/templates/types/models.tmpl

```diff
@@ -0,0 +1,14 @@
+package provider
```

Comment:
Do we want everything to be generated within a single go package?

(I don't know the answer to this, I've never written a Terraform provider)

Reply:
I will move them into their own packages.